radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/model/CoreModelMapper.java

```diff
@@ -0,0 +1,1207 @@
+import com.radixdlt.constraintmachine.REProcessedTxn;
+import com.radixdlt.constraintmachine.REStateUpdate;
+
+import com.radixdlt.constraintmachine.exceptions.SubstateNotFoundException;
+import com.radixdlt.crypto.ECDSASignature;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.crypto.exception.PublicKeyException;
+import com.radixdlt.engine.FeeConstructionException;
+import com.radixdlt.engine.RadixEngineException;
+import com.radixdlt.engine.parser.exceptions.TxnParseException;
+import com.radixdlt.identifiers.AID;
+import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.ledger.AccumulatorState;
+import com.radixdlt.mempool.MempoolFullException;
+import com.radixdlt.network.p2p.PeersView;
+import com.radixdlt.networks.Addressing;
+import com.radixdlt.networks.Network;
+import com.radixdlt.networks.NetworkId;
+import com.radixdlt.statecomputer.forks.ForkConfig;
+import com.radixdlt.statecomputer.forks.Forks;
+import com.radixdlt.statecomputer.forks.RERulesConfig;
+import com.radixdlt.utils.Bytes;
+import com.radixdlt.utils.Pair;
+import com.radixdlt.utils.PrivateKeys;
+import com.radixdlt.utils.UInt256;
+import com.radixdlt.utils.UInt384;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+public final class CoreModelMapper {
+	private static final String TOKEN_TYPE = "Token";
+	private static final String STAKE_UNIT_TYPE = "StakeUnit";
+	private static final String SYSTEM_ADDRESS = "system";
+	private static final String PREPARED_STAKES_ADDRESS = "prepared_stakes";
+	private static final String PREPARED_UNSTAKES_ADDRESS = "prepared_unstakes";
+	private static final String EXITING_UNSTAKES_ADDRESS = "exiting_unstakes";
+	private static final ECPublicKey MOCK_PUBLIC_KEY = PrivateKeys.ofNumeric(1).getPublicKey();
+	private final Addressing addressing;
+	private final Network network;
+	private final Forks forks;
+
+	@Inject
+	CoreModelMapper(
+		@NetworkId int networkId,
+		Addressing addressing,
+		Forks forks
+	) {
+		this.network = Network.ofId(networkId).orElseThrow();
+		this.addressing = addressing;
+		this.forks = forks;
+	}
+
+	public void verifyNetwork(NetworkIdentifier networkIdentifier) throws CoreApiException {
+		if (!networkIdentifier.getNetwork().equals(this.network.name().toLowerCase())) {
+			throw CoreApiException.notSupported(
+				new NetworkNotSupportedError()
+					.addSupportedNetworksItem(new NetworkIdentifier().network(this.network.name().toLowerCase()))
+					.type(NetworkNotSupportedError.class.getSimpleName())
+			);
+		}
+	}
+
+	public Pair<ECPublicKey, ECDSASignature> keyAndSignature(Signature signature) throws CoreApiException {
+		var bytes = bytes(signature.getBytes());
+		ECDSASignature sig;
+		try {
+			sig = ECDSASignature.decodeFromDER(bytes);
+		} catch (IllegalArgumentException e) {
+			throw CoreApiException.badRequest(
```

Comment:
Should we add or log the original exception? Do you there are more cases like that in other parts of the code?

Reply:
No, in this case, this is more of a user input exception. If anything we could add more of the original exception message to the response but I think it's fine without it for now.